### PR TITLE
[Listing] RESOPS27 - Set up ci for the listings service

### LINKED ISSUE #27: Set up CI for the listings service
## Feature Description
Create a GH CI Workflow that runs on every `push` for the listings service that runs: linting, formatting, testing, security

## Problem You're Trying to Solve
We want to enforce that only code that "passes" tests/lints/etc are merged. Creating a CI workflow will ensure that "bad" code does not get merged and doesn't disrupt other developers.

### PR DESCRIPTION
**Pull request by TobySMurray**
# Description

Create the base CI jobs for the Listing service: linting, formatting, testing, and security

This PR is stacked on top of #101, so first review that PR (as this diff contains that PR diff also).

Closes #27 

## How to Test

View the CI output below or run the actions locally with [act](https://github.com/nektos/act)

## Checklist
- [x] The code includes tests if relevant
- [x] I have *actually* self-reviewed my changes and done QA

### CODE REVIEW

File: .github/workflows/search.yml
Comment (kylestang) [inline .github/workflows/search.yml:1]: Do we want to be deleting this?

File: docker-compose.yml
Comment (kylestang) [inline docker-compose.yml:1]: This is replacing the search service in the docker compose. Don't we want to add the listing service alongside it?

**Review by kylestang — CHANGES_REQUESTED** [review]
Looks good except I don't think we want to be removing the search service config

**Review by Scott-Kenning — APPROVED** [review]

**Review by kylestang — APPROVED** [review]
LGTM

**Pull request merged**